### PR TITLE
User login

### PR DESCRIPTION
**Pull request by hubert-hernoga**

### CODE REVIEW

File: OnlineShop/urls.py [hunk 1]
```diff
@@ -32,8 +32,8 @@
     url('checkout_shipping/', CheckoutShipping.as_view(), name='checkout_shipping'),
     url('checkout_review/', CheckoutReview.as_view(), name='checkout_review'),
     url('checkout_complete/', CheckoutComplete.as_view(), name='checkout_complete'),
+    url('registration_account/', AccountRegistration.as_view(), name='registration_account'),
     url('account_login/', AccountLogin.as_view(), name='account_login'),
-    # url('index/', ProductsList.as_view(), name='index'),
     # url('account_orders/', AccountOrders.as_view(), name='account_orders'),
```
Comment (pawel-krysiak) [inline OnlineShop/urls.py]: @wezymord What about those routes? They dissapeared?

File: online_shop/migrations/0008_auto_20180322_2117.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        ('online_shop', '0007_auto_20180322_2113'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='profile',
+            name='user',
+            field=models.OneToOneField(default=None, on_delete=django.db.models.deletion.CASCADE, related_name='profile', to=settings.AUTH_USER_MODEL),
+        ),
+    ]
```
Comment (pawel-krysiak) [inline online_shop/migrations/0008_auto_20180322_2117.py]: @wezymord All new migrations again?

File: online_shop/templates/account-login.html [hunk 3]
```diff
@@ -26,75 +26,142 @@ <h1>Login / Register Account</h1>
...
-                  <input class="custom-control-input" type="checkbox" id="remember_me" checked>
-                  <label class="custom-control-label" for="remember_me">Remember me</label>
-                </div><a class="navi-link" href="account-password-recovery.html">Forgot password?</a>
-              </div>
-              <div class="text-center text-sm-right">
-                <button class="btn btn-primary margin-bottom-none" type="submit">Login</button>
-              </div>
+
+
+            <form action="/account_login/" class="login-box" method="post">
```
Comment (pawel-krysiak) [inline online_shop/templates/account-login.html]: 👍

File: online_shop/views.py [hunk 4]
```diff
@@ -169,28 +171,95 @@ def get(self, request):
...
         shipping_method_id = request.session['shipping_method']['shipping_method_id']
         shipping_method = ShippingOption.objects.get(pk=shipping_method_id)
 
-        user = NewUser(first_name=order['first_name'], last_name=order['last_name'], e_mail=order['email'],
-                       phone_number=order['phone_number'], company=order['company'], country=order['country'],
-                       city=order['city'], postal_code=order['postal_code'], address1=order['address1'],
-                       address2=order['address2'])
+        user = User(username=order['first_name'], first_name=order['first_name'],
+                    last_name=order['last_name'], email=order['email'])
+
```
Comment (pawel-krysiak) [inline online_shop/views.py]: we first get all users in ` users = User.objects.all()` and then create new one ?
every get request on `CheckoutComplete` ?

File: online_shop/views.py [hunk 5]
```diff
@@ -169,28 +171,95 @@ def get(self, request):
...
+        for user in users:
+            user.profile.phone_number = order['phone_number']
+            user.profile.company = order['company']
+            user.profile.country = order['country']
+            user.profile.city = order['city']
+            user.profile.postal_code = order['postal_code']
+            user.profile.address1 = order['address1']
+            user.profile.address2 = order['address2']
+            user.save()
+
```
Comment (pawel-krysiak) [inline online_shop/views.py]: and then for **every** user in the system, we assign attributes from the currently made order? That way every order we modify every user in the system, am I correct?

File: online_shop/views.py [hunk 6]
```diff
@@ -169,28 +171,95 @@ def get(self, request):
...
+        for user in users:
+            user.profile.phone_number = user_form['phone_number']
+            user.profile.company = user_form['company']
+            user.profile.country = user_form['country']
+            user.profile.city = user_form['city']
+            user.profile.postal_code = user_form['postal_code']
+            user.profile.address1 = user_form['address1']
+            user.profile.address2 = user_form['address2']
+            user.save()
```
Comment (pawel-krysiak) [inline online_shop/views.py]: We overwrite every existing user data?

File: online_shop/templates/account-login.html [hunk 7]
```diff
@@ -26,75 +26,142 @@ <h1>Login / Register Account</h1>
...
+                    <input class="form-control" name="email" placeholder="Email" required><span class="input-group-addon"><i class="icon-mail"></i></span>
+                  </div>
+                  <div class="form-group input-group">
+                    <input class="form-control" type="password" name="password" placeholder="Password" required><span class="input-group-addon"><i class="icon-lock"></i></span>
+                  </div>
+                  <div class="d-flex flex-wrap justify-content-between padding-bottom-10x">
+                    <div class="custom-control custom-checkbox">
+                      <input class="custom-control-input" type="checkbox" id="remember_me" checked>
+                      <label class="custom-control-label" for="remember_me">Remember me</label>
+                    </div><a class="navi-link" href="account-password-recovery.html">Forgot password?</a>
```
Comment (pawel-krysiak) [inline online_shop/templates/account-login.html]: Password recovery works now or no?

File: OnlineShop/urls.py (hunk 1 above)
Comment (wezymord) [reply OnlineShop/urls.py]: This was the first version. You probably mean 
``    url('show_all_products/', ShowAllProducts.as_view(), name='show_all_products')  ``

File: online_shop/migrations/0008_auto_20180322_2117.py (hunk 2 above)
Comment (wezymord) [reply online_shop/migrations/0008_auto_20180322_2117.py]: Nope, not again, we talked about it ;)

File: online_shop/views.py (hunk 4 above)
Comment (wezymord) [reply online_shop/views.py]: Yes.
Good point but first assigns  `` User.objects.all() `` to a variable `` users `` then create new user and at last I use this variable (when the user is already saved in the database). It works

File: online_shop/views.py (hunk 5 above)
Comment (wezymord) [reply online_shop/views.py]: 1. Nope, we assign attributes from the currently made order only for one user but I'm guessing it works only on one browser. I will make it work properly.
2. Yep, you're right. I just fixed this.

File: online_shop/views.py (hunk 6 above)
Comment (wezymord) [reply online_shop/views.py]: Not any more ;)

File: online_shop/templates/account-login.html (hunk 7 above)
Comment (wezymord) [reply online_shop/templates/account-login.html]: Not yet. I'm working on it

File: online_shop/views.py (hunk 4 above)
Comment (pawel-krysiak) [reply online_shop/views.py]: @wezymord but why? This is a serious error. We don't want all of the users to be overwritten each time somebody goes to the checkout page, do we? I think we dont even want to create user in the checkout page.

I guess we need to create it before when the guy is on the first step.

The logic is the following:
1. Guy has basked with chosen products
2. He clicks checkout
3. He lands on the first step with data to fill. ( address, email, etc... )
On that page he has choice either to click log in, if he has account, or...  fill all the form with necessary data and go to the second step.

If he logs in -> You redirect him back to the first step with pre-filled data from database.
If he does not log in, fill all the form and goes to next step -> You send post request, check if the user is in DB ( database ). and if you find user with this email you redirect him to login page and when he logs in you redirect him to **first step** , if you don't find one, you create account and move him to the **next step**

File: OnlineShop/urls.py (hunk 1 above)
Comment (pawel-krysiak) [reply OnlineShop/urls.py]: I just see 2 comments and i don't know why those 2 lines are commented out.

File: online_shop/templates/account-login.html (hunk 7 above)
Comment (pawel-krysiak) [reply online_shop/templates/account-login.html]: Move it to next PR

File: OnlineShop/urls.py (hunk 1 above)
Comment (wezymord) [reply OnlineShop/urls.py]: First I removed because we didnt use it, the second one is not used yet. But soon will be ;)

**Pull request merged**